pkg/executor/chained_task_handle.go

```diff
@@ -0,0 +1,119 @@
+package executor
+
+import (
+	"sync"
+	"time"
+)
+
+// ChainedTaskHandle is an links Launchers in a way that
+// one will be launched after another.
+type ChainedTaskHandle struct {
+	TaskHandle
+
+	chainedLaunchers []Launcher
+
+	encounteredError error
+
+	chainFinished chan struct{}
+	stopChain     chan struct{}
+	stopOnce      sync.Once
+}
+
+// NewChainedTaskHandle returns TaskHandle that executes current handle, and will
+// launch Launcher when handle will finish it's execution.
+func NewChainedTaskHandle(handle TaskHandle, launcher ...Launcher) TaskHandle {
+	launchers := make([]Launcher, 0)
```

Comment:
@skonefal - I don't think that we need this to iterate over `launcher`; it is slice of `executor.Launcher` anyway,